g2p/cli.py

```diff
@@ -700,34 +701,64 @@ def update(in_dir, out_dir):
 def update_schema(out_dir):
     """Generate a schema for the model configuration.
 
-    This should only be done once for each Minor version.
+    This should be done every time the model changes in a way that affects the schema.
     Changes to the schema should result in a minor version bump.
+    But not every minor or major version bump requires a schema update.
     """
     # Defer expensive imports
     from g2p.mappings import MappingConfig
 
-    # We should not be changing the schema for patches, so only include major/minor version
-    MAJOR_MINOR_VERSION = ".".join(VERSION.split(".")[:2])
+    # We shall not change the schema for patches, so only include major/minor version
+    (major, minor, *_rest) = g2p._version.version_tuple
+    major_minor = f"{major}.{minor}"
 
     # Determine path
     if out_dir is None:
-        schema_path = (
-            Path(LANGS_DIR) / f"../.schema/g2p-config-schema-{MAJOR_MINOR_VERSION}.json"
-        )
+        out_dir = Path(LANGS_DIR).parent / ".schema"
     else:
-        schema_path = Path(out_dir) / f"g2p-config-schema-{MAJOR_MINOR_VERSION}.json"
+        out_dir = Path(out_dir)
+    schema_path = out_dir / f"g2p-config-schema-{major_minor}.json"
+
     # Generate schema
-    if schema_path.exists():
-        raise FileExistsError(
-            f"Sorry a schema already exists for version {MAJOR_MINOR_VERSION}. "
-            "Please bump the minor version number and generate the schema again."
-        )
     json_schema = MappingConfig.model_json_schema()
-    # Add explicit schema dialect for SchemaStore
-    # Note that pydantic actually targets
+    # Add explicit schema dialect for SchemaStore that pydantic actually targets
     json_schema["$schema"] = "http://json-schema.org/draft-07/schema#"
-    with open(schema_path, "w") as f:
-        json.dump(json_schema, f, indent=2)
+
+    if schema_path.exists():
+        with open(schema_path, encoding="utf8") as f:
+            old_schema = json.load(f)
+        if old_schema == json_schema:
+            print(f"Schema {schema_path} is already up to date.")
+        else:
+            print(
+                dedent(
+                    f"""
+                    Schema {schema_path}
```

Comment:
black isn't complaining about these line lengths?

Reply:
Adjusted a bit, but they're sized to look decent after dedenting, so they're going to stay somewhat long.
I agree they were too long, though, so I adjusted them somewhat.